# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmMemory.h
# Commit: 85d47a33200c
# Full Hash: 85d47a33200c36a5f91f011f5713c34da79329ee
# Author: Julien Pages <jpages@mozilla.com>
# Date: 2025-09-15 21:44:14
# Regressor Bug: 1969509
# File Overlap Count: 7
# Overlapping Files: js/src/vm/ArrayBufferObject.cpp, js/src/wasm/WasmMemory.h, js/src/jit-test/tests/wasm/wasm-resizablearraybuffer-shared.js, js/src/jit-test/tests/wasm/memory64/basic.js, js/src/jit-test/tests/wasm/wasm-resizablearraybuffer.js
# Description:
#   Bug 1969509 - wasm: Update .maxByteLength to report the source wasm max pages in bytes. r=rhunt
#   
#   Following a spec change, 64-bit memory limits in wasm have been increased.
#   It's now possible for a wasm memory to have a limit bigger than the maximum
#   value of a 32-bits integer. To avoid an overflow, the maxByteLength getters
# ==============================================================================

diff -r 5860a981c750 -r 85d47a33200c js/src/wasm/WasmMemory.h
--- a/js/src/wasm/WasmMemory.h	Mon Sep 15 13:51:31 2025 +0000
+++ b/js/src/wasm/WasmMemory.h	Mon Sep 15 14:01:27 2025 +0000
@@ -102,6 +102,13 @@
     return length.value();
   }
 
+  // Return the byteLength for a 64-bits memory.
+  uint64_t byteLength64() const {
+    mozilla::CheckedInt<uint64_t> length(value_);
+    length *= PageSize;
+    return length.value();
+  }
+
   // Increment this pages by delta and return whether the resulting value
   // did not overflow. If there is no overflow, then this is set to the
   // resulting value.
